src/Services/Webhooks/Jwks.php

```diff
@@ -0,0 +1,150 @@
+use TrueLayer\Interfaces\Webhooks\JwksInterface;
+use TrueLayer\Services\Api\WebhooksApi;
+
+class Jwks implements JwksInterface
+{
+    private const CACHE_TTL = 1800;
+
+    private ApiClientInterface $api;
+
+    private ?EncryptedCacheInterface $cache;
+
+    private ValidatorFactory $validatorFactory;
+
+    private ?array $keys = null;
+
+    private ?int $retrievedAt = null;
+
+    public function __construct(ApiClientInterface $api,
+                                ?EncryptedCacheInterface $cache,
+                                ValidatorFactory $validatorFactory)
+    {
+        $this->api = $api;
+        $this->cache = $cache;
+        $this->validatorFactory = $validatorFactory;
+    }
+
+    /**
+     * @throws ApiResponseUnsuccessfulException
+     * @throws SignerException
+     * @throws ValidationException
+     * @throws ApiRequestJsonSerializationException
+     *
+     * @return array
+     */
+    public function getJsonKeys(): array
+    {
+        if (!$this->keys) {
+            if ($this->cache && $this->cache->has(CacheKeys::JWKS_KEYS)) {
+                $data = $this->cache->get(CacheKeys::JWKS_KEYS);
+
+                $this->keys = $data['keys'];
+                $this->retrievedAt = $data['retrieved_at'];
+            } else {
+                $this->retrieve();
+            }
+        }
+
+        if ($this->isExpired()) {
+            $this->retrieve();
+        }
+
+        return $this->keys;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getRetrievedAt(): ?int
+    {
+        return $this->retrievedAt ?? null;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isExpired(): bool
+    {
+        if (!$this->getRetrievedAt()) {
+            return true;
+        }
+
+        return $this->getRetrievedAt() + self::CACHE_TTL <= Carbon::now()->timestamp;
+    }
+
+    /**
+     * @return JwksInterface
+     */
+    public function clear(): JwksInterface
+    {
+        $this->keys = null;
+        $this->retrievedAt = null;
+
+        return $this;
+    }
+
+    /**
+     * @throws ApiResponseUnsuccessfulException
+     * @throws ApiRequestJsonSerializationException
+     * @throws SignerException
+     * @throws ValidationException
+     */
+    private function retrieve(): void
+    {
+        $data = (new WebhooksApi($this->api))->jwks();
+        $this->validate($data);
+
+        $this->keys = $data['keys'];
+        $this->retrievedAt = Carbon::now()->timestamp;
+
+        if ($this->cache) {
+            $this->cache->set(CacheKeys::JWKS_KEYS, $this->toArray(), self::CACHE_TTL);
+        }
+    }
+
+    /**
+     * @param array $data
+     *
+     * @throws ValidationException
+     */
+    private function validate(array $data): void
+    {
+        $validator = $this->validatorFactory->make($data, [
+            'keys.*.kty' => 'required|string|in:RSA,EC',
+            'keys.*.alg' => 'required|string|in:RS512,EC512',
```

Comment:
I think this will make the lib error out if new keys are released in different algos. Should we avoid validating here and maybe filtering the list depending on type?